desktop/ui/src/main/java/org/datacleaner/connection/DatastoreDescriptorDesktopBindings.java

```diff
@@ -0,0 +1,186 @@
+            "Connect to a PostgreSQL database", JdbcDatastore.class);
+
+    private static final DatastoreDescriptor ORACLE_DATASTORE_DESCRIPTOR = new DatastoreDescriptorImpl(DatabaseDriverCatalog.DATABASE_NAME_ORACLE,
+            "Connect to a Oracle database", JdbcDatastore.class);
+
+    private static final DatastoreDescriptor SQLSERVER_DATASTORE_DESCRIPTOR = new DatastoreDescriptorImpl(DatabaseDriverCatalog.DATABASE_NAME_MICROSOFT_SQL_SERVER_JTDS,
+            "Connect to a Microsoft SQL Server database", JdbcDatastore.class);
+    
+    private static final DatastoreDescriptor COMPOSITE_DATASTORE_DESCRIPTOR = new DatastoreDescriptorImpl("Composite datastore",
+            "Create a composite datastore", CompositeDatastore.class);
+
+    private static Map<DatastoreDescriptor, String> _iconPaths = new HashMap<DatastoreDescriptor, String>();
+
+    private static Map<DatastoreDescriptor, Class<? extends AbstractDatastoreDialog<? extends Datastore>>> _dialogClasses = new HashMap<>();
+
+    static {
+        _iconPaths.put(CSV_DATASTORE_DESCRIPTOR, IconUtils.CSV_IMAGEPATH);
+        _iconPaths.put(ACCESS_DATASTORE_DESCRIPTOR, IconUtils.ACCESS_IMAGEPATH);
+        _iconPaths.put(EXCEL_DATASTORE_DESCRIPTOR, IconUtils.EXCEL_IMAGEPATH);
+        _iconPaths.put(SAS_DATASTORE_DESCRIPTOR, IconUtils.SAS_IMAGEPATH);
+        _iconPaths.put(DBASE_DATASTORE_DESCRIPTOR, IconUtils.DBASE_IMAGEPATH);
+        _iconPaths.put(FIXEDWIDTH_DATASTORE_DESCRIPTOR, IconUtils.FIXEDWIDTH_IMAGEPATH);
+        _iconPaths.put(XML_DATASTORE_DESCRIPTOR, IconUtils.XML_IMAGEPATH);
+        _iconPaths.put(JSON_DATASTORE_DESCRIPTOR, IconUtils.JSON_IMAGEPATH);
+        _iconPaths.put(SALESFORCE_DATASTORE_DESCRIPTOR, IconUtils.SALESFORCE_IMAGEPATH);
+        _iconPaths.put(SUGARCRM_DATASTORE_DESCRIPTOR, IconUtils.SUGAR_CRM_IMAGEPATH);
+        _iconPaths.put(MONGODB_DATASTORE_DESCRIPTOR, IconUtils.MONGODB_IMAGEPATH);
+        _iconPaths.put(COUCHDB_DATASTORE_DESCRIPTOR, IconUtils.COUCHDB_IMAGEPATH);
+        _iconPaths.put(ELASTICSEARCH_DATASTORE_DESCRIPTOR, IconUtils.ELASTICSEARCH_IMAGEPATH);
+        _iconPaths.put(CASSANDRA_DATASTORE_DESCRIPTOR, IconUtils.CASSANDRA_IMAGEPATH);
+        _iconPaths.put(HBASE_DATASTORE_DESCRIPTOR, IconUtils.HBASE_IMAGEPATH);
+        _iconPaths.put(HIVE_DATASTORE_DESCRIPTOR, "images/datastore-types/databases/hive.png");
+        _iconPaths.put(MYSQL_DATASTORE_DESCRIPTOR, "images/datastore-types/databases/mysql.png");
+        _iconPaths.put(POSTGRESQL_DATASTORE_DESCRIPTOR, "images/datastore-types/databases/postgresql.png");
+        _iconPaths.put(ORACLE_DATASTORE_DESCRIPTOR, "images/datastore-types/databases/oracle.png");
+        _iconPaths.put(SQLSERVER_DATASTORE_DESCRIPTOR, "images/datastore-types/databases/microsoft.png");
+        _iconPaths.put(COMPOSITE_DATASTORE_DESCRIPTOR, IconUtils.COMPOSITE_IMAGEPATH);
+    }
+
+    static {
+        _dialogClasses.put(CSV_DATASTORE_DESCRIPTOR, CsvDatastoreDialog.class);
```

Comment:
... and then you can also add the dialog class direcly on the datastore descriptor instead of having this additional map.